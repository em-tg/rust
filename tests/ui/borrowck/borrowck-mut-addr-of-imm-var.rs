//@ check-pass
fn main() {
    let x: isize = 3;
    let y: &mut isize = &mut x; //~ WARNING cannot borrow
    *y = 5;
    println!("{}", *y);
}
