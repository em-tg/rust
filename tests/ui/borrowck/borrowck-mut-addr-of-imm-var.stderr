warning[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
  --> $DIR/borrowck-mut-addr-of-imm-var.rs:4:25
   |
LL |     let y: &mut isize = &mut x;
   |                         ^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut x: isize = 3;
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
